awpy/visualization/plot.py

```diff
@@ -326,3 +341,382 @@ def plot_nades(
+            (either with or without players)
+        given_fig_ax: Optional tuple containing figure and ax objects for plotting
+        save_filepath (str): Filepath to save the figure to file if required
+
+    Returns: Nothing, all plotting is done on ax object
+
+    Raises:
+        ValueError: If map_name is not in awpy.data.NAV
+    """
+    if map_name not in NAV:
+        msg = "Map not found."
+        raise ValueError(msg)
+
+    player_positions = extract_player_positions(frame)
+    map_control_dict = calc_map_control_values(map_name, player_positions)
+    if plot_type == "players":
+        plot_map_control_snapshot(
+            map_name,
+            map_control_dict,
+            player_pos=player_positions,
+            given_fig_ax=given_fig_ax,
+            save_filepath=save_filepath,
+        )
+    else:
+        plot_map_control_snapshot(
+            map_name,
+            map_control_dict,
+            given_fig_ax=given_fig_ax,
+            save_filepath=save_filepath,
+        )
+
+
+def create_round_map_control_gif(
+    map_name: str,
+    round_data: GameRound,
+    plot_type: str = "",
+    gif_filepath: str = "./results/round_mc.gif",
+) -> None:
+    """Create gif summarizing map control for round.
+
+    Args:
+        map_name (str): Map used in plot_frame_map_control call
+        round_data (GameRound): Round whose map control will be animated.
+            Expected format that of awpy round
+        plot_type (str): Determines which type of plot is created
+            (either with or without players)
+        gif_filepath (str): Filepath to save the gif to file
+
+    Returns: Nothing, all plotting is done on ax object
+
+    Raises:
+        ValueError: If map_name is not in awpy.data.NAV
+                    If roundData not of expected
+                    awpy frame format
+    """
+    if map_name not in NAV:
+        msg = "Map not found."
+        raise ValueError(msg)
+    if "frames" not in round_data:
+        msg = "round_data argument not of expected awpy round format."
+        raise ValueError(msg)
+    if "tmp" not in os.listdir():
+        error_string = (
+            "Frame visualizations cannot be saved to the"
+            "./tmp folder - it does not exist"
+        )
+        raise ValueError(error_string)
+
+    images: list[np.ndarray] = []
+    print("Saving/loading frames")
+    frames = round_data["frames"]
+    i = 0
+    while frames and i < len(frames):
+        frame = frames[i]
+        filename = f"./tmp/tmp{i}.png"
+
+        # Save current frame map control viz to file
+        # All frames are saved to './tmp/ folder '
+        plot_frame_map_control(
+            map_name,
+            frame,
+            plot_type=plot_type,
+            save_filepath=filename,
+        )
+
+        # Load image back as frame of gif that will
+        # be created at the end of this function
+        images.append(imageio.imread(filename))
+        i += 1
+
```

Comment:
This should probably be a for loop with enumerate instead of a while loop.

Reply:
Yeah I changed this to a while because I was having pyright issues a week ago and had no idea had to fix it so changed the for loop to a while as a last ditch effort. Changed in back in next commit to branch.